test/k8s_tests/domain_mapping_e2e_test.go

```diff
@@ -0,0 +1,126 @@
+package k8s_tests
+
+import (
+	mock "github.com/dana-team/container-app-operator/test/k8s_tests/mocks"
+	utilst "github.com/dana-team/container-app-operator/test/k8s_tests/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	knativev1beta1 "knative.dev/serving/pkg/apis/serving/v1beta1"
+)
+
+var _ = Describe("Validate DomainMapping functionality", func() {
+
+	It("Should create DomainMapping when Creating a Capp instance", func() {
+		By("Creating a capp with a route")
+		routeCapp := mock.CreateBaseCapp()
+		routeHostname := utilst.GenerateRouteHostname()
+		routeCapp.Spec.RouteSpec.Hostname = routeHostname
+		utilst.CreateCapp(k8sClient, routeCapp)
+		By("Checking if the domainMapping was created successfully")
+		domainMapping := &knativev1beta1.DomainMapping{
```

Comment:
you can transfer the object creation to the mocks and pass it name and namespace